### PR TITLE
Fix class toggle component to only toggle class for intended target

### PR DESCRIPTION
**Pull request by restlessbit**
There's a handy JS component called `.js-class-toggle` that lets you toggle a CSS class for an element from markup.

It works like this:

```html
<!-- Clicking this button will toggle the "beep--boop" class on the div -->
<button class="js-class-toggle" data-target="#someTarget" data-class="beep---boop">Bop it</button>

<div id="someTarget" class="beep">Twist it</div>
```

There was one small problem with it, which is that it will toggle all elements with the required `.js-class-toggle`, even if that element wasn't the intended target.

I hope that explanation made sense...

/cc @underdogio/engineering

### CODE REVIEW

File: scripts/components/class-toggle.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 function createClassToggle (element) {
-  var target = document.querySelectorAll(element.dataset.target)[0];
+  var targetSelector = element.dataset.target;
+  var target = document.querySelectorAll()[0];
```
Comment (brettlangdon) [inline scripts/components/class-toggle.js]: missing `targetSelector` here? `document.querySelectorAll(targetSelector)[0]`?

Comment (brettlangdon) [inline scripts/components/class-toggle.js]: also, why not `document.querySelector()` instead of `querySelectorAll()[0]`

File: scripts/components/class-toggle.js [hunk 2]
```diff
@@ -8,7 +9,13 @@ function createClassToggle (element) {
   var classToToggle = element.dataset.class;
 
   document.addEventListener('click', function (e) {
-    if (e.target.classList.contains('js-class-toggle') || e.target.closest('.js-class-toggle')) {
+    var eventTarget = e.target;
+    var closestParent = eventTarget.closest('.js-class-toggle');
+
+    // Toggle class if an element with the .js-class-toggle was clicked, and if
+    // the clicked element's target matches the target for this element
+    if ((eventTarget.classList.contains('js-class-toggle') || closestParent) && (
```
Comment (brettlangdon) [inline scripts/components/class-toggle.js]: `.matches` is nice.

`eventTarget.matches('.js-class-toggle')`

File: scripts/components/class-toggle.js (hunk 1 above)
Comment (restlessbit) [inline scripts/components/class-toggle.js]: woops 👊

File: scripts/components/class-toggle.js (hunk 2 above)
Comment (brettlangdon) [inline scripts/components/class-toggle.js]: This might be more clear if it was written as multiple if statements.

I am still not 100% sure what is going on here. I mean, I get the point, but hard to follow the nested `(` and `)` and `||` and `&&` everywhere.

**Comment by restlessbit** [comment]
Changes made. I think I made that conditional a little less confusing 🎱

**Comment by restlessbit** [comment]
Oh I opted to not use `.matches` because apparently Edge and IE don't support it without a prefix 🙄

**Comment by brettlangdon** [comment]
ugh... edge....
